### ISSUE TITLE
Can't connect to an IPv6 DB from the app container

### ISSUE THREAD

**Issue by mattgibson** [basecamp/kamal#1322]
I've just tried setting up Rails on a cloud server via Kamal and providing connection details to a free supabase DB, which is IPv6 only. I get a `Network Unreachable` error from ActiveRecord on deploy, and the same happens when I use psql from inside the container. If I SSH into the instance and use psql outside the container, it connects correctly.

Not sure if https://github.com/basecamp/kamal/pull/1280 or something similar will fix it

**Comment by HLFH** [comment]
@mattgibson Yes, my PR https://github.com/basecamp/kamal/pull/1280 fixes it.

**Comment by jorgegonzalez** [comment]
Getting the same issue, anyone have a workaround yet?

**Comment by disbelief** [comment]
Seems both PRs were closed without merging, is ipv6 supported yet or not?

**Comment by djmb** [comment]
We're not going to add that by default because it might not work everywhere.

Instead you can create the kamal network yourself. Kamal doesn't try to create it if it already exists.

You could do it in a pre-deploy hook to ensure it is always created as you want it:

```shell
# .kamal/hooks/pre-deploy
set -e

docker network create --ipv6 kamal || true
```

**Comment by disbelief** [comment]
@djmb are there default network settings that we should apply to maintain the network configuration that a standard kamal installation would expect or is `docker network create kamal` all that is needed?

**Comment by jorgegonzalez** [comment]
> We're not going to add that by default because it might not work everywhere.
> 
> Instead you can create the kamal network yourself. Kamal doesn't try to create it if it already exists.
> 
> You could do it in a pre-deploy hook to ensure it is always created as you want it:

Okay but it would be nice to have this in the docs

**Comment by karlentwistle** [comment]
Just ran into this issue as well. I followed this guide to get it working on an already deployed app: https://klevo.sk/unix/enable-ipv6-connectivity-from-containers-deployed-with-kamal/
